### DISCUSSION TITLE
Add support for app-check

### DISCUSSION THREAD

**Discussion by smohri-michibiki** [kreait/firebase-php#618] [Ideas]
about Firebase app-check
https://firebase.google.com/docs/app-check

Firebase-admin-node already support app-check mechanism like below.
https://github.com/firebase/firebase-admin-node/tree/master/src/app-check

I want to use this mechanism at my php application (API server). I guess app-check implementation on php is same like Firebase Auth and JWT token verifier.

Thank you.

**Comment by NafaaAzaiez** [comment]
That would be awesome !

**Comment by jeromegamez** [comment]
The current version of the PHP SDK, unfortunately, has a lot of legacy under the hood, which makes it quite difficult to add new functionality, especially in the Auth component.

I'm currently working on a 6.0 release (you can see the progress in the 6.x branch) which, after 5 years without really breaking changes will introduce some.

When that is done, I can think about adding new features again, like AppCheck Token and Session Cookie Verification, but I have no ETA of the new release yet.

Please keep in mind that I'm a single developer and not part of Google. Without GitHub sponsors or companies hiring me to work on the SDK (I can provide invoices ^^), I can only work on it in my spare time and in my own pace. 

I do appreciate you using the SDK and your feature request and promise that it is and stays on my todo list, I hope you understand that/why I can not promise _when_ it will be done 🙏
